src/info/infra/repositories/DataverseInfoRepository.ts

```diff
@@ -4,17 +4,19 @@ import { ReadError } from '../../../core/domain/repositories/ReadError';
 import { DataverseVersion } from '../../domain/models/DataverseVersion';
 
 export class DataverseInfoRepository implements IDataverseInfoRepository {
-  private apiUrl: string;
-
-  constructor(apiUrl: string) {
-    this.apiUrl = apiUrl;
-  }
+  constructor(private readonly apiUrl: string) {}
 
   public async getDataverseVersion(): Promise<DataverseVersion> {
-    const response = await axios.get(`${this.apiUrl}/info/version`).catch((error) => {
-      throw new ReadError(error.response.status + error.response.data ? ': ' + error.response.data.message : '');
-    });
-    return this.getVersionFromResponse(response);
+    let datasetVersion;
+    await axios
+      .get(`${this.apiUrl}/info/version`)
+      .then((response) => {
+        datasetVersion = this.getVersionFromResponse(response);
+      })
+      .catch((error) => {
+        throw new ReadError(error.response.status + error.response.data ? ': ' + error.response.data.message : '');
+      });
+    return datasetVersion;
```

Comment:
```suggestion
    return await axios
      .get(`${this.apiUrl}/info/version`)
      .then((response) => this.getVersionFromResponse(response))
      .catch((error) => throw new ReadError(error.response.status + error.response.data ? ': ' + error.response.data.message : ''));
```

If you are not going to use the datasetVersion for other operations you can directly do the return right?

Reply:
The intention was to make it more visual. But I agree, I have simplified it as you propose.

Reply:
Your intention makes all the sense to me. But I think JS developers are more used to this kind of `return function.then.then.catch` syntax than to putting the return at the end

I don't know if that makes for you, is more javascript, like when people say is more pythonic but with JS, is more javascripthonic (?

Thanks for accepting the change!